### PR TITLE
WAL-2484: Set JWT in cookie

### PR DESCRIPTION
**Pull request by maxisch**
## Description

Set the JWT in a cookie for the passkeys server to consume.

Crossbit-main PR: https://github.com/Paella-Labs/crossbit-main/pull/13853

## Test plan

Creating a user in the scw_demo page.

### CODE REVIEW

File: packages/client/wallets/aa/src/blockchain/wallets/passkey.ts [hunk 1]
```diff
@@ -103,4 +105,10 @@ export class PasskeyWalletService {
             type: "passkeys",
         };
     }
+
+    private setJwtCookie(jwt: string) {
+        const now = new Date();
+        const expireTime = new Date(now.getTime() + 1 * 60 * 60 * 1000); // Now + 1 hour
+        document.cookie = `user_jwt=${jwt}; expires=${expireTime.toUTCString()}; path=/`;
+    }
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/passkey.ts:113]: If we're not running on localhost, we should use the `secure` property here.

File: packages/client/wallets/aa/src/blockchain/wallets/passkey.ts [hunk 2]
```diff
@@ -103,4 +105,10 @@ export class PasskeyWalletService {
             type: "passkeys",
         };
     }
+
+    private setJwtCookie(jwt: string) {
+        const now = new Date();
+        const expireTime = new Date(now.getTime() + 1 * 60 * 60 * 1000); // Now + 1 hour
+        document.cookie = `user_jwt=${jwt}; expires=${expireTime.toUTCString()}; path=/`;
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/passkey.ts:112]: If you don't specify it, this cookie will only be sent to servers w/ the domain of the client that set it. When `harrypotter.com` uses SDK, `crossmint.com` won't receive the cookies.

File: packages/client/wallets/aa/src/blockchain/wallets/passkey.ts [hunk 3]
```diff
@@ -103,4 +105,10 @@ export class PasskeyWalletService {
             type: "passkeys",
         };
     }
+
+    private setJwtCookie(jwt: string) {
+        const now = new Date();
+        const expireTime = new Date(now.getTime() + 1 * 60 * 60 * 1000); // Now + 1 hour
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/passkey.ts:111]: maybe worth defining a const `JWT_COOKIE_EXPIRY_TIME` or something like that.

**Review by devlyn37 — APPROVED** [review]
Left a couple important comments to check out before landing!

File: packages/client/wallets/aa/src/blockchain/wallets/passkey.ts (hunk 2 above)
Comment (maxisch) [reply packages/client/wallets/aa/src/blockchain/wallets/passkey.ts:112]: I read a bit on this, what options do we have? We can't set cookies cross domains. Should we set the cookie server side (for crossmint.com) on the passkeys server on let's say the first interaction?

Comment (devlyn37) [reply packages/client/wallets/aa/src/blockchain/wallets/passkey.ts]: That sounds like a good idea to me, maybe we could have some "handshake" endpoint on the passkeys controller to set the cookie

**Comment by maxisch** [comment]
Closing as we are not moving forward with this approach

**Pull request closed without merging**